core/src/ffi/mod.rs

```diff
@@ -139,3 +142,35 @@ impl ToResult for raw::c_int {
         }
     }
 }
+
+/// Makes NonNull even easier to use with the downside that it hides the
+/// unsafeness of the underlying pointer access.
+pub(crate) struct EasyPtr<T>(NonNull<T>);
```

Comment:
Is there a better alternative to investigate here?

Reply:
probably not? this is completely internal to capsule. it just cuts down on how many `unsafe {}` we have to sprinkle through out the code base and the use is completely limited to the ffi layer. for example, all the packet headers are held as `NonNull`, not `EasyPtr`. if you see any usage outside the ffi layer, call it out.

Reply:
Will double check